scripts/deseq2.R

```diff
@@ -425,17 +425,17 @@ input.csv <- paste(out,"input.csv",sep="")
 write.csv(as.data.frame(df), file=input.csv)
 
 if (length(patients) > 0) {
-    sampleTable <- data.frame(sampleName = samples, fileName = samples, condition = conditions, type = col.labels, patients = patients)
+    sampleTable <- data.frame(sampleName = samples, fileName = samples, condition = conditions, type = col.labels, patients = patients, design= patients + conditions)
     # ddsHTSeq <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable, directory = "", design= ~ patients + condition) # doesn"t work with nextflow
     ddsHTSeq <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable, design= ~ patients + condition)
 } else {
-    sampleTable <- data.frame(sampleName = samples, fileName = samples, condition = conditions, type = col.labels)
+    sampleTable <- data.frame(sampleName = samples, fileName = samples, condition = conditions, type = col.labels, design= conditions)
```

Comment:
@flomock can you explain why adding `design = conditions` here? And does this fix your initial issue with the script?

Reply:
Ah, sorry i forgot to mention.
I added here a further design column which can than be used in the PCA plot (line 478).
This should than result in the PCA plot I would expect as user. Meaning that when I provide a condition all samples with the same condition will have the same color code. If I provide conditions and patients then all sample which are from the same patient AND condition have the same color.

The previous behavior was that each sample as a unique color. But what you want to see in this plot is if the samples, you would expect, cluster together. There is no need to identify each replicate by an unique color as the x,y coordinates by itself don't tell you anything. The information is rather in the placement of the sample comparison to the other samples.

At least this is my opinion. So my version holds less information (dont now which point is which replicat) but is much clearer and thus easier to understand.

Reply:
old version
[https://www.dropbox.com/s/axevfvvhh8orctc/pca_simple.pdf?dl=0](url)
new version
[https://www.dropbox.com/s/s81y8rd1a8ha9s3/pca_simple_new.pdf?dl=0](url)

Further note that in my version vsd rather than rld is used for the pca.
Simply because this is the way they deseq2 authors do it in the tutorial ([http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#principal-component-plot-of-the-samples](url))

Reply:
Okay, regarding PCA I can't totally agree. In many cases, you do this PCA to identify outliers and remove them from your analysis. And in such cases, you need to know the replicate. 

But I agree that the PCA plotting is not optimal (and the plots look shitty anyway). It's more to get the first idea of your sample and then I would recommend anyway to re-do the PCA. (that's what I developed PCAGO back than, ... but yeah... you know). 

I think we can go with your version for now and I will simply merge your pull request and then tweaking this version of the R script in a separate branch. 

thanks for digging into this!